cmd/config/docs/api-conventions/config-io.md

```diff
@@ -1,65 +0,0 @@
-# Configuration IO API Semantics
```

Comment:
Although I appreciate the explicit descriptions of how this applies to configuration functions within that spec, this document was broader. Reading through it, it seems to document how e.g. kyaml itself should operate. The description of `local-config` has also disappeared entirely with this PR. Can we restore this file please?

Reply:
This is a valid suggestion, I'll follow up. I think there are two categories of annotations:

- "control annotations": Annotations that are used for orchestration of functions, but are otherwise not persisted to disk. Examples: "path", "index", "id" (Used in comment preservation logic). Ideally, we want to reserve a particular annotation prefix to enable control annotations to be added in the future without requiring changes to all functions. If this is not "config.kubernetes.io", we need to reserve something else, e.g. "fn.config.kubernetes.io".
- Other annotations that are persisted to disk. Example: "local-config".

I think larger issue is that there's no clarify on what prefix annotations should have. Even in SIG-CLI projects, there are couple of annotation prefixes.